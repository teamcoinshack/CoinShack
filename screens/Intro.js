import React from 'react';
import { 
  StyleSheet,
  View,
  Text,
  Image,
  Dimensions,
} from 'react-native';
import AppIntroSlider from 'react-native-app-intro-slider';
import Firebase from 'firebase';
import { background, textHeader, textSubheader } from '../Masterlist';

const slides = [
  {
    key: '0',
    isImage: false,
    image: require('../assets/intro/wallet.png'),
  },
  {
    key: '1',
    isImage: true,
    image: require('../assets/intro/wallet.png'),
  },
  {
    key: '2',
    isImage: true,
    image: require('../assets/intro/market.png'),
  },
  {
    key: '3',
    isImage: true,
    image: require('../assets/intro/news.png'),
  },
  {
    key: '4',
    isImage: true,
    image: require('../assets/intro/profile.png'),
  },
  {
    key: '5',
    isImage: true,
    image: require('../assets/intro/social.png'),
  },
];

export default class Intro extends React.Component {
  _renderItem = item => {
    if (item.isImage) {
      return (
        <View style={styles.slide}>
          <Image 
            resizeMode="contain"
            style={{
              height: Dimensions.get('window').height,
              width: Dimensions.get('window').width,
            }}
            source={item.image} 
          />
        </View>
      );
    } else {
      return (
        <View style={styles.welcome}>
          <Image
            source={require('../assets/icons/CoinShackIcon.png')}
            style={{
              height: 100,
              width: 100,
              margin: 10
            }}
          />
          <Text style={styles.header}>
            Welcome to CoinShack!
          </Text>
          <Text style={styles.content}> {/* hol can you do the short write up for this part lmao how to trade crypto */}
            Cryptocurrency trading is easy! Just buy low and sell high! Our prices are real time! Learn as you trade!
          </Text>
          <Text style={styles.content}>
            Swipe to take a quick tour of our app!
          </Text>
        </View>
      );
    }
  }

  _onDone = () => {
    // User finished the introduction. Show real app through
    // navigation or simply by controlling state
    Firebase.auth().currentUser.isNew = false;
    this.props.navigation.navigate('Dashboard');
  }

  render() {
    return (
      <AppIntroSlider 
        renderItem={this._renderItem} 
        slides={slides} 
        renderDoneButton={this._renderDoneButton}
        renderNextButton={this._renderNextButton}
        onDone={this._onDone}
        showSkipButton
      />
    );
  }
}

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'black',
  },
  welcome: {
    backgroundColor: background,
    justifyContent: 'center',
    alignItems: "center",
    flex: 1,
  },
  header: {
    textAlign: 'center',
    color: textHeader,
    fontSize: 22,
    fontWeight: "600"
  },
  content: {
    textAlign: 'center',
    color: textSubheader,
    fontSize: 18,
    margin: 10,
  }
})
