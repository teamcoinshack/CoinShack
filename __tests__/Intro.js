/**
 * @format
 */

import 'react-native';
import React from 'react';
import Intro from '../screens/Intro.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-google-signin', () => {});
test('renders correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});
