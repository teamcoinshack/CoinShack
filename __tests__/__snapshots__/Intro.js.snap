// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "content1": "A risk-free simulator for anyone to learn to trade cryptocurrency!",
          "content2": "With our app, you will receive free virtual currency for you to start trading with real-time market data for numerous cryptocurrencies without any risk involved!",
          "content3": "Swipe to learn more about cryptocurrency trading and our simulator system.",
          "header": "Welcome to CoinShack!",
          "key": "0",
        },
        Object {
          "content1": "You just received $10,000 worth of in game cash for you to purchase any cryptocurrency coins!",
          "content2": "The in game cash and prices of coins are based on real-time USD rates. You can see the amount of cash and cryptocurrencies you own and trade them in the Wallet tab.",
          "content3": "You can also look at the trends of the different cryptocurrencies in the Market tab. Stay updated to the latest news in the News tab and compete with your friends in the Social tab. Happy trading!",
          "header": "Quickstart Guide",
          "key": "1",
        },
      ]
    }
    disableVirtualization={false}
    extraData={750}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    paginationStyle={null}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showDoneButton={true}
    showNextButton={true}
    showsHorizontalScrollIndicator={false}
    slides={
      Array [
        Object {
          "content1": "A risk-free simulator for anyone to learn to trade cryptocurrency!",
          "content2": "With our app, you will receive free virtual currency for you to start trading with real-time market data for numerous cryptocurrencies without any risk involved!",
          "content3": "Swipe to learn more about cryptocurrency trading and our simulator system.",
          "header": "Welcome to CoinShack!",
          "key": "0",
        },
        Object {
          "content1": "You just received $10,000 worth of in game cash for you to purchase any cryptocurrency coins!",
          "content2": "The in game cash and prices of coins are based on real-time USD rates. You can see the amount of cash and cryptocurrencies you own and trade them in the Wallet tab.",
          "content3": "You can also look at the trends of the different cryptocurrencies in the Market tab. Stay updated to the latest news in the News tab and compete with your friends in the Social tab. Happy trading!",
          "header": "Quickstart Guide",
          "key": "1",
        },
      ]
    }
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#373b48",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/icons/CoinShackIcon.png",
                }
              }
              style={
                Object {
                  "height": 100,
                  "margin": 10,
                  "width": 100,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#dbdbdb",
                  "fontSize": 22,
                  "fontWeight": "600",
                  "textAlign": "center",
                }
              }
            >
              Welcome to CoinShack!
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              A risk-free simulator for anyone to learn to trade cryptocurrency!
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              With our app, you will receive free virtual currency for you to start trading with real-time market data for numerous cryptocurrencies without any risk involved!
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              Swipe to learn more about cryptocurrency trading and our simulator system.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#373b48",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/icons/CoinShackIcon.png",
                }
              }
              style={
                Object {
                  "height": 100,
                  "margin": 10,
                  "width": 100,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#dbdbdb",
                  "fontSize": 22,
                  "fontWeight": "600",
                  "textAlign": "center",
                }
              }
            >
              Quickstart Guide
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              You just received $10,000 worth of in game cash for you to purchase any cryptocurrency coins!
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              The in game cash and prices of coins are based on real-time USD rates. You can see the amount of cash and cryptocurrencies you own and trade them in the Wallet tab.
            </Text>
            <Text
              style={
                Object {
                  "color": "#a4a9b9",
                  "fontSize": 18,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              You can also look at the trends of the different cryptocurrencies in the Market tab. Stay updated to the latest news in the News tab and compete with your friends in the Social tab. Happy trading!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 16,
          "left": 16,
          "position": "absolute",
          "right": 16,
        },
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 16,
          "justifyContent": "center",
          "margin": 16,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, .9)",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 10,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, .2)",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 18,
                "padding": 12,
              },
              null,
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
